cli/src/commands/auth/commands/login.ts

```diff
@@ -45,16 +47,43 @@ export default (opts: BaseCommandOptions) => {
       program.error('Could not perform authentication. Please try again');
     }
 
-    const organizations = new Set(decoded.groups.map((group) => group.split('/')[1]));
+    const organizationSlugs = new Set(decoded.groups.map((group) => group.split('/')[1]));
 
+    const organizationSlugByDisplayKey = new Map<string, string>();
+    for (const organizationSlug of organizationSlugs) {
+      const headers = getBaseHeaders();
+      const response = await opts.client.platform.getOrganizationBySlug(
+        {
+          slug: organizationSlug,
+        },
+        {
+          headers: {
+            ...headers,
+            authorization: `Bearer ${accessTokenResp.response.accessToken}`,
+            'cosmo-org-slug': organizationSlug,
+          },
+        },
+      );
+      if (!response.response || response.response.code !== EnumStatusCode.OK || !response.organization) {
+        organizationSlugByDisplayKey.set(organizationSlug, organizationSlug);
+        continue;
+      }
+      organizationSlugByDisplayKey.set(`${response.organization.name} (${organizationSlug})`, organizationSlug);
+    }
     const selectedOrganization = await inquirer.prompt({
-      name: 'organizationSlug',
+      name: 'organizationKey',
       type: 'list',
       message: 'Select Organization:',
-      choices: [...organizations],
+      choices: [...organizationSlugByDisplayKey.keys()],
```

Comment:
I could save a second iteration by producing an array when adding the keys themselves